init.go

```diff
@@ -93,165 +89,121 @@ func initConfigDefaults() {
+		return nil, nil, fmt.Errorf("unable to get apid instance info: %v", err)
 	}
 
 	if config.IsSet(configApidInstanceID) {
 		log.Warnf("ApigeeSync plugin overriding %s.", configApidInstanceID)
 	}
 	config.Set(configApidInstanceID, apidInfo.InstanceID)
 
-	return nil
-}
+	apidTokenManager := createApidTokenManager(apidInfo.IsNewInstance)
+	var snapMan snapshotManager
+	var apidChangeManager changeManager
 
-func createManagers() {
 	if isOfflineMode {
-		apidSnapshotManager = &offlineSnapshotManager{}
+		snapMan = &offlineSnapshotManager{
+			dbMan: apidDbManager,
+		}
 		apidChangeManager = &offlineChangeManager{}
 	} else {
-		apidSnapshotManager = createSnapShotManager()
-		apidChangeManager = createChangeManager()
-	}
-
-	apidTokenManager = createSimpleTokenManager()
-}
-
-func checkForRequiredValues() error {
-	required := []string{configProxyServerBaseURI, configConsumerKey, configConsumerSecret}
-	if !isOfflineMode {
-		required = append(required, configSnapServerBaseURI, configChangeServerBaseURI)
-	}
-	// check for required values
-	for _, key := range required {
-		if !config.IsSet(key) {
-			return fmt.Errorf("Missing required config value: %s", key)
+		httpClient := &http.Client{
+			Transport: tr,
+			Timeout:   httpTimeout,
+			CheckRedirect: func(req *http.Request, _ []*http.Request) error {
+				req.Header.Set("Authorization", "Bearer "+apidTokenManager.getBearerToken())
+				return nil
+			},
 		}
-	}
-	proto := config.GetString(configSnapshotProtocol)
-	if proto != "sqlite" {
-		return fmt.Errorf("Illegal value for %s. Only currently supported snashot protocol is sqlite", configSnapshotProtocol)
+		snapMan = createSnapShotManager(apidDbManager, apidTokenManager, httpClient)
+		apidChangeManager = createChangeManager(apidDbManager, snapMan, apidTokenManager, httpClient)
 	}
 
-	return nil
-}
+	listenerMan := &listenerManager{
+		changeMan:     apidChangeManager,
+		snapMan:       snapMan,
+		tokenMan:      apidTokenManager,
+		isOfflineMode: isOfflineMode,
+	}
 
-func SetLogger(logger apid.LogService) {
-	log = logger
+	apiMan := &ApiManager{
+		endpoint: tokenEndpoint,
+		tokenMan: apidTokenManager,
+	}
+	return listenerMan, apiMan, nil
 }
 
-/* initialization */
-func _initPlugin(services apid.Services) error {
+func initPlugin(services apid.Services) (apid.PluginData, error) {
+	SetLogger(services.Log().ForModule("apigeeSync"))
+	dataService = services.Data()
+	eventService = services.Events()
+	apiService = services.API()
 	log.Debug("start init")
-
 	config = services.Config()
 	initConfigDefaults()
 
+	isOfflineMode := false
 	if config.GetBool(configDiagnosticMode) {
 		log.Warn("Diagnostic mode: will not download changelist and snapshots!")
 		isOfflineMode = true
 	}
 
-	err := checkForRequiredValues()
+	err := checkForRequiredValues(isOfflineMode)
 	if err != nil {
-		return err
+		return PluginData, err
 	}
-
-	err = initVariables()
 	if err != nil {
-		return err
+		return PluginData, err
```

Comment:
In case of `err` what would be value for that `PluginData`?

Reply:
Still the same, this is not a problem because `PluginData` is hard-coded in code.